pkg/test/local/envoy/discovery/filter.go

```diff
@@ -0,0 +1,181 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package discovery
+
+import (
+	"context"
+	"io"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	ads "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	// Identity is a FilterFunc that always returns the original discovery response, unaltered.
+	Identity = func(resp *xdsapi.DiscoveryResponse) (*xdsapi.DiscoveryResponse, error) {
+		return resp, nil
+	}
+)
+
+// FilterFunc is a function that allows an application to filter Envoy discovery responses
+type FilterFunc func(resp *xdsapi.DiscoveryResponse) (*xdsapi.DiscoveryResponse, error)
+
+// Filter proxies requests to an Envoy discovery server (e.g. Pilot) and filters the responses.
+type Filter struct {
+	// DiscoveryAddr is the address of the discovery server
```

Comment:
Perhaps make it explicit that this is the address of Filter's discovery service.

Reply:
I'm curious what would make this more obvious? The property is already named `DiscoveryAddr`, which seems pretty straight-forward.  Would `DiscoveryServiceAddr` be better in your mind?

Reply:
Discussed offline and modified the comment to clarify.